File: Proyecto2/app/src/main/java/com/PMultimedia/proyecto2/MainActivity.java
package com.PMultimedia.proyecto2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button boton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        /*boton = findViewById(R.id.boton);
        boton.setOnClickListener(this);*/

        Log.i("Estado","Estoy en onCreate");

    }

    public void pulsado(View v) {
        finish();
    }

    public void onStart() {
        super.onStart();
        Log.i("Estado","Estoy en onStart");
    }

    public void onResume() {
        super.onResume();
        Log.i("Estado","Estoy en onResume");
    }

    public void onPause() {
        super.onPause();
        Log.i("Estado","Estoy en onPause");
    }

}

File: CustomViewTestCanvasFunctions/app/src/main/java/com/iesvirgendelcarmen/customview/MainActivity.kt
package com.iesvirgendelcarmen.customview

import android.graphics.Color
import android.graphics.Paint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {

    lateinit var customView: CustomView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        customView = findViewById<CustomView>(R.id.customView)
//        customView.paint = Paint().apply {
//            color = Color.RED
//        }

//        val boton = findViewById<Button>(R.id.button)
//        boton.setOnClickListener {
//            customView.paint = Paint().apply {
//                color = Color.GREEN
//            }
//        }

    }
}
